packages/wrangler/src/codemod.ts

```diff
@@ -0,0 +1,225 @@
+import assert from "node:assert";
+import { existsSync } from "node:fs";
+import path from "node:path";
+import { UserError } from "@cloudflare/workers-utils";
+import * as recast from "recast";
+import { transformFile } from "./autoconfig/c3-vendor/codemod";
+import { createCommand } from "./core/create-command";
+import { logger } from "./logger";
+
+const b = recast.types.builders;
+const t = recast.types.namedTypes;
+
+function isNamedProp(
+	prop: unknown,
+	name: string
+): prop is
+	| recast.types.namedTypes.ObjectProperty
+	| recast.types.namedTypes.Property {
+	return (
+		(t.Property.check(prop) || t.ObjectProperty.check(prop)) &&
+		t.Identifier.check(prop.key) &&
+		prop.key.name === name
+	);
+}
+
+function vitestPoolV3ToV4(args: string | undefined) {
+	const filePathTS = path.join(process.cwd(), `vitest.config.ts`);
+	const filePathJS = path.join(process.cwd(), `vitest.config.js`);
+
+	let filePath: string;
+
+	if (args) {
+		if (!existsSync(args)) {
+			throw new UserError(
+				`Selected Vitest config file does not exist: '${args}'`
+			);
+		}
+		filePath = path.join(process.cwd(), args);
+	} else if (existsSync(filePathTS)) {
+		filePath = filePathTS;
+	} else if (existsSync(filePathJS)) {
+		filePath = filePathJS;
+	} else {
+		throw new UserError("Could not find Vitest config file to modify");
+	}
+
+	logger.log(filePath);
+
+	transformFile(filePath, {
```

Comment:
It would be great if there was a dry run option or confirm step. Is this something it can do?

Reply:
Good shout!